[ENH] Derived (processed) MR data (#109)
bids-standard/bids-specification: src/99-appendices/08-coordinate-systems.md

# Appendix VIII: preferred names of Coordinate systems

## Definitions

### Coordinate System

A coordinate system specifies the information necessary to interpret the
coordinates associated with the data. A coordinate system can be described by
its topology (cartesian, spherical, polar, etc.,.), dimensionality, and units of
measurement along each dimension. Every device that makes spatial measurements
or has sensors described in a space has a coordinate system (e.g., an MR

> scanner, an MEG scanner, an EEG cap, Polhemus digitizer). The data output by

Reviewer: an EEG cap is not a spatial measurement device and has no intrinsic coordinates. It is the Polhemus that measures coordinates of electrodes.

Reviewer 2: Could you suggest a better wording (roping in @satra for additional comments).

Reviewer 3: @robertoostenveld - better wording would be great. but as far as i know the cap is used by people to refer to coordinates of channels, without doing any digitization - https://en.wikipedia.org/wiki/10%E2%80%9320_system_(EEG)

Reviewer: @satra you are confusing "electrode placement system" (the design of the schema) with "electrode cap" (the flexible fabric). The 10-20 system does not depend on a cap, but does require measurements. To make electrode application faster, researchers often skip the detailed measurement but use a cap (with only a few measurements) instead. See also 
https://doi.org/10.1016/S1388-2457(00)00527-7

Reviewer 2: From my current reading of the contents of this PR the section "Definitions" in Appendix VIII that this discussion revolves around is not crucial for clarifying the use of any of the data types described elsewhere in the PR. Perhaps we could move it to a new PR and work on it on an independent schedule?

most of these devices describe the number of dimensions, the units along each
dimension, and in some cases the details of origin (for example, time).

### Space, atlas and map

> A space|atlas|map is embedded in a coordinate system to ascribe meaning to the

Reviewer 4: I don't think we use `|` elsewhere:

```suggestion
A space/atlas/map is embedded in a coordinate system to ascribe meaning to the
```

> coordinates. For instance, Scanner, Talairach88, MNI, FS are different

Reviewer 5: Here it is mentioned that 'Scanner' is a coordinate system, but it is not listed in the label below Volume. Should we use 'Other' to describe scanner coordinates?

Reviewer: I agree that "scanner" is not a good name. Elsewhere we use "other" and then require the detailled specification in the _coordsystem.json file. And [elsewhere](
https://github.com/bids-standard/bids-specification/pull/109#issuecomment-471099335)  "orig" is used.

I would consider "device" a good name. It generalizes to different devices and toes not make implicit assumptions about temporal order (as is done in "orig", which is prone to be interpreted differently by different people).

Reviewer 2: From my current reading of the contents of this PR the section "Definitions" in Appendix VIII that this discussion revolves around is not crucial for clarifying the use of any of the data types described elsewhere in the PR. Perhaps we could move it to a new PR and work on it on an independent schedule?

Reviewer: As a comment from the meta-level: I must admit that I find it very hard to keep the of the PRs goals and content on github (in general, also this one).  I also fear that others with even more limited github skills than me might not be able to contribute. 

The changes in this PR relate to specification texts  that are not restricted to (f)MRI. That is why I am commenting to `derivatives/01-introduction.md` (which will in the future also apply to meg/ieeg/eeg), common-data-types (it does not define scope and does not say "MR specific data types", so I guess I'll have to deal with this in the future as well), and a "space/coordinates" appendix that underlies not only derivatives, but also the raw data specification and that actually originated from MEG and now gets used for MRI derivatives as well. 

I would not object if those three (or two, in case you consider document `02` to be `mr-specific-data-types`) were discussed in their own PRs.

Reviewer 2: Yeah, it's challenging. We have been using smaller PRs to this PR to structure work, but it's not perfect. I don't think that splitting introduction, the few necessary changes to Appendix VIII, and actual new data types (MR derivatives) into separate independently mergeable PRs would be good. The new data types drive the "common" rules, and the "common" rules need to work with the new data types. Those need to be considered together. Of course, it also sets the ground for future derivatives - that's why your input is so important and why I appreciate you monitor this PR. The sections you focus on are the best use of your time.

What I was proposing in the previous comment was to shrink the scope of this PR by moving the "Definitions" section of https://github.com/bids-standard/bids-specification/blob/c42c4bbb9a84886740e8c2583575ae68363c2d15/src/99-appendices/08-coordinate-systems.md to a separate PR. Its content does not influence much of the rest of the PR and it seems to be causing some controversy that can be resolved on a different schedule.

Reviewer 5: Being clear about how to refer to scanner space does seem relevant for the current data type (MRI/DTI). For example: fiber pathways have x,y,z coordinates, per these derivatives, do they always have to be defined in a normalized/ACPC space or could they potentially live in scanner space? 
(The fact that we also define iEEG electrode coordinates as the center of mass of objects from a T1w image is a side-note, but we do want to write code to integrate DTI/MRI/iEEG data)

Reviewer 3: at least in MR the coordinate system is encoded (somewhat) in the nifti header of the raw data. `space-orig` was intended to capture that a derivative contains information in the same "space" as the original file from which it was derived. if there is an intermediate alignment step, then the ReferenceMap would point to a realigned file.

Reviewer 2: Indeed - I wasn't clear. Specifying what coordinate system derivatives are is important and we are working hard on refining this language (see for example https://github.com/chrisfilo/bids-specification/pull/12/files), However the extra "definitions" section in Appendix VIII is not strictly necessary for this purpose (as it wasn't for `space` in context of iEEG). I think it's a useful addtion, but could be discussed in a separate PR.

Reviewer 5: Thanks for clarifying - is it then the case that ReferenceMap is similar to how we use IntendedFor in _coordsystem.json, and the <label> to use in space-<label> for scanner space is 'Other'?
(I would not like to start naming my ieeg files _space-Other and then other derived data _space-SomethingElseToDescribeScannerSpace)

Reviewer 2: to keep the threads on topic I replied at https://github.com/chrisfilo/bids-specification/pull/12#issuecomment-473985746

coordinate systems, while the Talairach atlas, MNI152, and fsaverage ascribe
meaning to the coordinates in terms of brain anatomy. As an example to
distinguish coordinate systems and spaces/atlas/maps, a 4D fMRI timeseries
acquired without motion correction has individual time points all in the same
coordinate system, but the atlas/map is different (due to motion) in different
time points.

### Landmarks

Landmarks are properties of the object being measured and can exist in any
coordinate system in which the object is placed/measured. They can be relevant
points of the imaged object's structure (i.e. anatomical landmarks) such as the

> anterior commissure in a brain. They can also be devices that generate a

Reviewer: Devices that generate a localizable signal are "fiducials" or "fiducial markers", not "landmarks" . See https://en.wikipedia.org/wiki/Fiducial_marker and also the MEG and EEG specifications.

Reviewer 3: how would you describe a vitamin-e pill, or an mr-safe deep-brain stimulation device? the signals are passive, not active, like the MEG signal generators.

> anterior commissure in a brain. They can also be devices that generate a

Reviewer 3: ```suggestion
anterior commissure in a brain. They can also be fiducials that generate a
```

localizable signal that are placed in the intended position before scanning
(i.e. markers). Positions of landmarks are identified via manual or automated

> algorithms that detect these landmarks in (e.g., MR) or from (e.g., MEG) the

Reviewer 3: ```suggestion
algorithms that detect these landmarks fiducials in (e.g., MR) or from (e.g., MEG) the
```

> algorithms that detect these landmarks in (e.g., MR) or from (e.g., MEG) the

Reviewer 3: ```suggestion
algorithms that detect these fiducials in (e.g., MR) or from (e.g., MEG) the
```

specified explicitly in the sidecar file in the `XXCoordinateSystemUnits` field,
the units are assumed to be mm.

### Volume

| Coordinate System                                   | Description                                                                                                                                                                                                                                                                                                                                                      |
| --------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | FS305                                               | FreeSurfer variant of the MNI305 space                                                                                                                                                                                                                                                                                                                           |

Reviewer 6: This should be implicit to `fsaverage*` for the same reasons of `MNI152NLin6AsymConte69` should not be a space. This example clearly shows how sampling should be detached from the notion of space.

Reviewer 2: This is just a label, but it should not include any special character (such as `*`), so if anything `fsaverage` can be used. I not familiar with this space though - @edickie did you add this one?

| MNI152Lin                                           | Also known as ICBM (version with linear coregistration) [http://www.bic.mni.mcgill.ca/ServicesAtlases/ICBM152Lin](http://www.bic.mni.mcgill.ca/ServicesAtlases/ICBM152Lin)                                                                                                                                                                                       |
| MNI152NLin6\[Sym&#124;Asym\]                        | Also known as ICBM 6th generation (non-linear coregistration). Used by SPM99 - SPM8 and FSL (MNI152NLin6Sym). [http://www.bic.mni.mcgill.ca/ServicesAtlases/ICBM152NLin6](http://www.bic.mni.mcgill.ca/ServicesAtlases/ICBM152NLin6)                                                                                                                             |

> | MNI152NLin6AsymConte69                              | Template based on MNI152NLin6Asym using Conte69 data. Built by Human Connectome Project. [https://github.com/Washington-University/HCPpipelines/tree/master/global/templates](https://github.com/Washington-University/HCPpipelines/tree/master/global/templates)                                                                                                |

Reviewer 6: IMHO This should be fsLR. Space has no notion of surfaces or volumes.

Reviewer 2: If you think `fsLR` for this volume space defined by a volume template would be more intuitive we can change it. Pinging @edickie for second opinion.

Reviewer 7: I dunno, I don't really know MNI152NLin6AsymConte69 (I actually just register to MNI152NLin6Sym in my workflows anyway..)

| FS305                                               | FreeSurfer variant of the MNI305 space                                                                                                                                                                                                                                                                                                                           |
| MNI152Lin                                           | Also known as ICBM (version with linear coregistration) [http://www.bic.mni.mcgill.ca/ServicesAtlases/ICBM152Lin](http://www.bic.mni.mcgill.ca/ServicesAtlases/ICBM152Lin)                                                                                                                                                                                       |

> | MNI152NLin6\[Sym&#124;Asym\]                        | Also known as ICBM 6th generation (non-linear coregistration). Used by SPM99 - SPM8 and FSL (MNI152NLin6Sym). [http://www.bic.mni.mcgill.ca/ServicesAtlases/ICBM152NLin6](http://www.bic.mni.mcgill.ca/ServicesAtlases/ICBM152NLin6)                                                                                                                             |

Reviewer 6: This is incorrect

```suggestion
| MNI152NLin6Sym                                      | Also known as ICBM152 6th generation (non-linear coregistration). [http://www.bic.mni.mcgill.ca/ServicesAtlases/ICBM152NLin6](http://www.bic.mni.mcgill.ca/ServicesAtlases/ICBM152NLin6)                                                                                                                             |
| FSLMNI152NLin6Asym                                  | A variation of the ``MNI152NLin6Sym`` template that is distributed and used by FSL.                                                                                                                                                  |
```

Refs.
https://github.com/templateflow/templateflow/issues/6#issuecomment-454648955
https://github.com/templateflow/templateflow/issues/6#issuecomment-455000224

The SPM bit might also be wrong. I'm not as positive here, but I think SPM has a custom derivation of the MNINLin6Sym template. Paging @gllmflndn here.

Reviewer 2: I don't think we can replace `MNI152NLin6Asym`  with `FSLMNI152NLin6Asym` due to backward compatibility, but we can add it if you want.

Reviewer 7: Ooooh. I was meaning to ask you guys about this! I got confused last week when the templateflow templates for fmriprep didn't work with my HCP pipeline steps (they use FSL)!

Reviewer 6: > I don't think we can replace `MNI152NLin6Asym`  with `FSLMNI152NLin6Asym` due to backward compatibility, but we can add it if you want.

I see, yes. I think a deprecation warning here would also be nice. I think it is crucial to call for attention on this detail.
